### ISSUE TITLE
Error: TypeError: Object of type 'int64' is not JSON serializable

### ISSUE THREAD

**Issue by joshidp** [Azure/MachineLearningNotebooks#1351]
Hi There,

I am running a forecasting task and getting below error.

**TypeError: Object of type 'int64' is not JSON serializable**

There is no Int64 type in my dataset, sample values are shown below.

1,A,4,12-28-2020,2020,12,28
2,A,4,12-29-2020,2020,12,29
2,A,4,12-31-2020,2020,12,31

Please suggest a resolution.

**Complete Error**

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup_AutoML_12eaeb1a-8b11-4591-aa1c-7bd05469f8a2.py", line 731, in <module>
    result = setup_run()
  File "setup_AutoML_12eaeb1a-8b11-4591-aa1c-7bd05469f8a2.py", line 725, in setup_run
    prep_type=preparation_type
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/train/automl/runtime/_remote_script.py", line 33, in setup_wrapper
    **kwargs)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/train/automl/runtime/_entrypoints/remote/setup_run.py", line 88, in execute
    run_lifecycle_utilities.fail_run(setup_run, e)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/automl/core/_exception_utilities.py", line 29, in wrapper
    return function(*args, **kwargs)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/automl/core/_run/run_lifecycle_utilities.py", line 140, in fail_run
    run.fail(error_details=interpreted_exception, error_code=error_code)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/core/run.py", line 2253, in fail
    super(_SubmittedRun, self).fail(error_details=error_details, error_code=error_code, _set_status=False)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/core/run.py", line 1350, in fail
    self._client.fail(error_details=error_details, error_code=error_code, _set_status=_set_status)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/_run_impl/run_history_facade.py", line 613, in fail
    self.set_failed_status(error_details=error_details, error_code=error_code, _set_status=_set_status)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/_run_impl/run_history_facade.py", line 616, in set_failed_status
    error_dict = self._error_details_to_dictionary(error_details=error_details, error_code=error_code)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/_run_impl/run_history_facade.py", line 508, in _error_details_to_dictionary
    error_dict = json.loads(error_details._serialize_json())['error']
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/site-packages/azureml/_common/exceptions.py", line 181, in _serialize_json
    return json.dumps(error_ret, indent=indent)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/azureml-envs/azureml_09ff55f546b313bb1ab136a466214499/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'int64' is not JSON serializable

Thanks

**Comment by cartacioS** [comment]
Hello @joshidp 

Could you please share your AutoML RunID so we can further investigate?

**Comment by cartacioS** [comment]
Hi @joshidp,

I used the runID in your logs so that the engineering team could take a closer look. First off, thanks for submitting the feedback! We have discovered an issue with how we raise an error in this situation- in actuality the issue is that one (or more) of your time series has only 1 unique target value, which is causing training to fail.

i.e. if you have sale quantity data for store_1, store_2, and store_3, say the `quantity` value for time stamps relating to store_1 are a constant value (unique value), this will cause failures. Hope that makes sense!

We are actively looking into how to better surface this error.

-Sabina
